### ISSUE TITLE
Serverless Framework@4.1.18 InitializingFailed

### ISSUE THREAD

**Issue by miniGau** [serverless/serverless#12721]
### Issue description

when running the `sls deploy` to deploy my service to AWS , it failed as follows:
```
TypeError: Invalid URL
    at new URL (node:internal/url:797:36)
    at getEndpointUrl (file:///Users/xx/.serverless/releases/4.1.18/package/dist/sf-core.js:76:36960)
    at evaluateEndpointRule (file:///Users/xx/.serverless/releases/4.1.18/package/dist/sf-core.js:76:37819)
```
my local serverless versions: 
Serverless ϟ Framework
 • 4.1.18
 
 it did work when it is in `4.1.15` yesterday.  it automatically upgrades to the latest version@4.1.18 which has not released yet.
 
 any help?
 many thanks.

**Comment by MatthewGlancy** [comment]
I believe I'm seeing the same issue:

```
Failed to construct URL with https://sts.${self:custom.envpath.region}.amazonaws.com/ TypeError: Invalid URL
    at new URL (node:internal/url:797:36)
    at getEndpointUrl (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:36960)
    at evaluateEndpointRule (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:37819)
    at evaluateRules (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:38586)
    at evaluateTreeRule (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:38355)
    at evaluateRules (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:38783)
    at evaluateTreeRule (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:38355)
    at evaluateRules (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:38783)
    at resolveEndpoint (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:76:39812)
    at Object.defaultEndpointResolver [as endpointProvider] (file:///home/runner/.serverless/releases/4.1.18/package/dist/sf-core.js:113:1[29](https://github.com/Go-Live-Data/engage/actions/runs/10176519853/job/28146835706#step:5:30)74) {
  code: 'ERR_INVALID_URL',
  input: 'https://sts.${self:custom.envpath.region}.amazonaws.com'
}
```

I was reading the project's region from my environment file.

serverless.yml BEFORE - doesn't work

```yml
custom:
  envpath: "${file(./serverless.env.yml):${opt:stage, self:provider.stage}}"
  
provider:
  name: aws
  runtime: nodejs20.x
  region: ${self:custom.envpath.region}
```

When I changed region to STOP reading from the custom.envpath then everything worked again:

serverless.yml AFTER - this does work at 4.1.18

```yml
custom:
  envpath: "${file(./serverless.env.yml):${opt:stage, self:provider.stage}}"
  
provider:
  name: aws
  runtime: nodejs20.x
  region: eu-west-1
```

I still have plenty other references throughout my serverless.yml - it was just the region that was causing the issue for me

**Comment by czubocha** [comment]
@miniGau @MatthewGlancy Thanks for reporting, we're looking into it.

**Comment by czubocha** [comment]
@miniGau @MatthewGlancy We released a fix in v4.1.19 that solves this issue. Thanks again for reporting.
